components/camel-spring/src/main/java/org/apache/camel/spring/SpringCamelContext.java

```diff
@@ -119,59 +119,85 @@ public static SpringCamelContext springCamelContext(String configLocations) thro
         return springCamelContext(new ClassPathXmlApplicationContext(configLocations));
     }
 
-    public void afterPropertiesSet() throws Exception {
-        StopWatch watch = new StopWatch();
-
-        maybeStart();
-        
-        LOG.debug("afterPropertiesSet() took {} millis", watch.stop());
-    }
-
-    public void destroy() throws Exception {
-        stop();
-    }
-
-    public void onApplicationEvent(ApplicationEvent event) {
-        LOG.debug("onApplicationEvent: {}", event);
+    @Override
+    public void start() {
+        // for example from unit testing we want to start Camel later (manually)
+        if (Boolean.TRUE.equals(NO_START.get())) {
+            LOG.trace("Ignoring maybeStart() as NO_START is false");
+            return;
+        }
 
-        if (event instanceof ContextRefreshedEvent) {
-            // now lets start the CamelContext so that all its possible
-            // dependencies are initialized
+        if (!isStarted() && !isStarting()) {
             try {
-                maybeStart();
+                StopWatch watch = new StopWatch();
+                super.start();
+                LOG.debug("start() took {} millis", watch.stop());
             } catch (Exception e) {
                 throw wrapRuntimeCamelException(e);
             }
-        } else if (event instanceof ContextClosedEvent) {
-            // ContextClosedEvent is emitted when Spring is about to be shutdown
-            if (isShutdownEager()) {
-                try {
-                    maybeStop();
-                } catch (Exception e) {
-                    throw wrapRuntimeCamelException(e);
-                }
-            }
-        } else if (event instanceof ContextStoppedEvent) {
-            // ContextStoppedEvent is emitted when Spring is end of shutdown
+        } else {
+            // ignore as Camel is already started
+            LOG.trace("Ignoring maybeStart() as Apache Camel is already started");
+        }
+    }
+
+    @Override
+    public void stop() {
+        if (!isStopping() && !isStopped()) {
             try {
-                maybeStop();
+                super.stop();
             } catch (Exception e) {
                 throw wrapRuntimeCamelException(e);
             }
+        } else {
+            // ignore as Camel is already stopped
+            LOG.trace("Ignoring maybeStop() as Apache Camel is already stopped");
```

Comment:
And btw we tend to write Camel instead of Apache Camel in these log messages, maybe we should just do that. Yeah its friday and its 'bike shed' day ;)